smack-extensions/src/main/java/org/jivesoftware/smackx/receipts/DeliveryReceipt.java

```diff
@@ -41,9 +41,14 @@
     private final String id;
 
     public DeliveryReceipt(String id) {
-        this.id = StringUtils.requireNotNullOrEmpty(id, "id must not be null");
+        this.id = id;
     }
 
+    /**
+     * Get the id of the message that has been delivered.
+     * Note: This method might return null!
+     * @return id of the delivered message
```

Comment:
Nit: Instead of the "Note: …" text, one does typical record the nullness of arguments and return values in the javadoc return or param description:
"id of the delivered message or {@code null}"

While possibly also mentioning the fact in the javadoc description.